src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -105,6 +105,39 @@ def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=T
         self._management_resource_uri = CLOUD.endpoints.management
         self._ad_resource_uri = CLOUD.endpoints.active_directory_resource_id
 
+    def refresh_subscriptions(self):
+        from azure.cli.core.adal_authentication import AdalAuthentication
+        from azure.mgmt.resource import SubscriptionClient
+        # If no subscription is currently set CLI will throw error asking user to set a subscription or login
+        subClient = SubscriptionClient(self.get_login_credentials()[0])
```

Comment:
Are we only refreshing the current azure account or across all of them? Please note, users could login with multiple accounts, and different service principals.